Updated prov schemas

MFSY opened the pull request:
- externalized the shacl context
     - externalized the standard context
     - merged usage schema in activity schema
     - added positive and negative tests for the prov schemas

Fixes #12 #13 #14

MFSY committed:
Updated prov schemas:
     - externalized the shacl context
     - externalized the standard context
     - merged usage schema in activity schema
     - added positive and negative tests for the prov schemas

annakristinkaufmann commented on modules/prov/src/main/resources/schemas/nexus/prov/collection/v0.1.0.json:
```diff
@@ -1,147 +1,26 @@
-          "description": "The (prov) entities that constitutes the collection. At least one prov:Entity should be provided.",
           "name": "Members",
-          "rdfs:label": "TODO: check how to enforce prov:Entity here",
+          "description": "The (prov) entities that constitutes the collection. At least one prov:Entity should be provided.",
```
should be "entities that constitute"

MFSY replied:
sure. [done](https://github.com/BlueBrain/nexus-prov/pull/15/commits/4e15e3c4e1feb5b5d4f070a18f81d769e72a0708)

MFSY committed:
Corrected minor typo in collection shape description

annakristinkaufmann commented on modules/prov/src/test/resources/data/nexus/prov/activity/v0.1.0/all-fields.json:
```diff
@@ -0,0 +1,71 @@
+  "used": [
+    {
+      "@type": "prov:Entity",
+      "@id": "{{base}}/data/bbp/experiment/experiment/specimen/v1.0.0/807ded58-579d-452d-8dd2-a5568689b2bb"
```
put subject instead of specimen

annakristinkaufmann commented on modules/prov/src/test/resources/data/nexus/prov/activity/v0.1.0/all-fields.json:
```diff
@@ -0,0 +1,71 @@
+        "@type": [
+          "prov:Entity"
+        ],
+        "@id": "{{base}}/data/bbp/experiment/experiment/specimen/v1.0.0/807ded58-579d-452d-8dd2-a5568689b2bb"
```
put subject instead of specimen

annakristinkaufmann commented on modules/prov/src/test/resources/invalid/nexus/prov/activity/v0.1.0/wrong-startedAtTimeValue.json:
```diff
@@ -0,0 +1,71 @@
+  "used": [
+    {
+      "@type": "prov:Entity",
+      "@id": "{{base}}/data/bbp/experiment/experiment/specimen/v1.0.0/807ded58-579d-452d-8dd2-a5568689b2bb"
```
replace specimen with subject

annakristinkaufmann commented on modules/prov/src/test/resources/invalid/nexus/prov/activity/v0.1.0/wrong-startedAtTimeValue.json:
```diff
@@ -0,0 +1,71 @@
+  "wasAssociatedWith": [
+    {
+      "@type": "prov:Agent",
+      "@id": "{{base}}/data/bbp/experiment/agent/v1.0.0/1c4d3683-dcfc-489f-adbb-5de815b1b89a"
```
put agent under core: data/bbp/core/agent

MFSY committed:
Fixed minor bugs
 - changed agent schema ref in activity schema
 - used subject instead of specimen
 - used slice instead of slice
